pints/_optimisers/_irpropmin.py

```diff
@@ -127,12 +166,35 @@ def needs_sensitivities(self):
 
     def n_hyper_parameters(self):
         """ See :meth:`pints.TunableMethod.n_hyper_parameters()`. """
-        return 0
+        return 2
 
     def running(self):
         """ See :meth:`Optimiser.running()`. """
         return self._running
 
+    def set_hyper_parameters(self, x):
+        """
+        See :meth:`pints.TunableMethod.set_hyper_parameters()`.
+
+        The hyper-parameter vector is ``[min_step_size, max_step_size]``.
+        """
+        self.set_min_step_size(x[0])
+        self.set_max_step_size(x[1])
+
+    def set_max_step_size(self, step_size):
+        """
+        Sets the maximum step size (use ``None`` to let step sizes grow
+        indefinitely).
+        """
+        self._step_max = None if step_size is None else float(step_size)
```

Comment:
Can we throw an error when step_max < step_min? Does the method break when step_max = step_min?

Reply:
😳  I didn't add it out of laziness.

Do you think we should validate late?
I wouldn't want this situation:

```
# Set min step size to 1, max to 10
opt.set_min_step_size(1)

ERROR! MAX STEP SIZE IS 0.1
```

so I was thinking the check would happen on the first call to `ask()` ?

Reply:
Hmm I don't have a strong opinion on this 😊 I personally think it's nice when attempting to set a hyperparameter that breaks the method immediately raises an error, or maybe at least a UserWarning? But if the error thrown in `ask()` is clear enough about which hyperparameter breaks the optimiser, I am also happy with that 😊